### PR TITLE
Refactoring / hardening related to the `Structural Injection` flag.

### PR DESCRIPTION
**Pull request by Zimmi48**

### CODE REVIEW

File: tactics/equality.ml [hunk 1]
```diff
@@ -1471,10 +1461,10 @@ let get_previous_hyp_position id gl =
   in
   aux MoveLast (Proofview.Goal.hyps gl)
 
-let injEq flags ?(old=false) with_evars clear_flag ipats =
+let injEq flags ?injection_in_context with_evars clear_flag ipats =
```
Comment (SkySkimmer) [inline tactics/equality.ml:1464]:
```suggestion
let injEq flags ?(injection_in_context=injection_in_context_flag ()) with_evars clear_flag ipats =
```

**Comment by Zimmi48** [comment]
The compatibility impact is too large to follow up on this idea, so I'm removing the last commit (changing the default value) from this PR. The rest can be considered clean-up / refactoring.

**Comment by SkySkimmer** [comment]
Ho are the stdlib changes related to the ocaml changes? Was it just to deal with the compat break from the removed commit?

**Comment by Zimmi48** [comment]
Yes, I could split this into two unrelated PRs if you want.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**